framework/aster-frame/src/trap/irq.rs

```diff
@@ -118,41 +118,62 @@ pub fn disable_local() -> DisabledLocalIrqGuard {
 
 /// A guard for disabled local IRQs.
 pub struct DisabledLocalIrqGuard {
-    was_enabled: bool,
+    /// Whether the local IRQs need to be re-enabled when this gaurd is dropped.
+    /// Usually depends on whether the local IRQs were enabled at the moment the guard was created.
+    need_reenable: bool,
     preempt_guard: DisablePreemptGuard,
 }
 
 impl !Send for DisabledLocalIrqGuard {}
 
+/// Disable local IRQs if they are enabled.
+/// Return whether the local IRQs were enabled before this function call.
+pub fn disable_irq_if_enabled() -> bool {
+    let was_enabled = irq::is_local_enabled();
+    if was_enabled {
+        irq::disable_local();
+    }
+    was_enabled
+}
+
+/// Enable local IRQs if they are disabled.
+/// Return whether the local IRQs were enabled before this function call.
+pub fn enable_irq_if_disabled() -> bool {
```

Comment:
When do we need this function? It seems that the return value of this function will never be used, at least in this PR. Is this function just added for consistency with `disable_irq_if_enabled`?